zio/parquetio/rowgroupfilter.go

```diff
@@ -0,0 +1,196 @@
+	zctx       *super.Context
+}
+
+func (r *rowGroupFilter) evalFilter(e dag.Expr) (match, ok bool) {
+	switch e := e.(type) {
+	case *dag.BinaryExpr:
+		switch e.Op {
+		case "and":
+			if match, ok := r.evalFilter(e.LHS); !match && ok {
+				return false, true
+			}
+			r.evalFilter(e.RHS)
+		case "or":
+			if match, ok := r.evalFilter(e.LHS); match || !ok {
+				return match, ok
+			}
+			return r.evalFilter(e.RHS)
+		case "in":
+			var elems []dag.VectorElem
+			switch e := e.RHS.(type) {
+			case *dag.ArrayExpr:
+				elems = e.Elems
+			case *dag.SetExpr:
+				elems = e.Elems
+			default:
+				return false, false
+			}
+			for _, elem := range elems {
+				vv, ok := elem.(*dag.VectorValue)
+				if !ok {
+					return false, false
+				}
+				if match, ok := r.evalComparison("==", e.LHS, vv.Expr); match || !ok {
+					return match, ok
+				}
+			}
+			return false, true
+		case "==", "!=", "<", "<=", ">", ">=":
+			return r.evalComparison(e.Op, e.LHS, e.RHS)
+		}
+	case *dag.UnaryExpr:
+		if e.Op == "!" {
+			match, ok := r.evalFilter(e.Operand)
+			return !match, ok
+		}
+	}
+	return false, false
+}
+
+func (r *rowGroupFilter) evalComparison(op string, lhs, rhs dag.Expr) (match, ok bool) {
+	if _, ok := lhs.(*dag.This); !ok {
+		if op[0] == '<' {
+			op = strings.ReplaceAll(op, "<", ">")
+		} else {
+			op = strings.ReplaceAll(op, ">", "<")
+		}
+		lhs, rhs = rhs, lhs
+	}
+	lhsThis, ok1 := lhs.(*dag.This)
+	rhsLiteral, ok2 := rhs.(*dag.Literal)
+	if !ok1 || !ok2 {
+		return false, false
+	}
+	col := r.schema.ColumnIndexByName(strings.Join(lhsThis.Path, "."))
+	if col < 0 {
+		return false, false
+	}
+	min, max, ok := columnChunkStats(r.rgMetadata, col)
+	if !ok {
+		return false, false
+	}
+	val, err := zson.ParseValue(r.zctx, rhsLiteral.Value)
+	if err != nil {
+		return false, false
+	}
+	minResult, ok1 := compare(val, min)
+	maxResult, ok2 := compare(val, max)
+	if !ok1 || !ok2 {
+		return false, false
+	}
+	switch op {
+	case "==":
+		// val >= min && val <= max
+		return minResult >= 0 && maxResult <= 0, true
+	case "!=":
+		// val < min || val > max
+		return minResult < 0 || maxResult > 0, true
+	case "<":
+		// val < max
+		return maxResult < 0, true
+	case "<=":
+		// val <= max
+		return maxResult <= 0, true
+	case ">":
+		// val > min
+		return minResult > 0, true
+	case ">=":
+		// val >= min
+		return minResult >= 0, true
+	}
+	panic(op)
+}
+
+func compare(aVal super.Value, b any) (int, bool) {
+	if aVal.IsNull() {
+		return 0, false
+	}
+	switch b := b.(type) {
+	case bool:
+		if aVal.Type().ID() != super.IDBool {
+			return 0, false
+		}
+		if a := aVal.Bool(); a == b {
+			return 0, true
+		} else if a {
+			return -1, true
+		}
+		return 1, true
+	case []byte:
+		if id := aVal.Type().ID(); id != super.IDBytes && id != super.IDString {
+			return 0, false
+		}
+		a := aVal.Bytes()
+		n := min(len(a), len(b))
+		return bytes.Compare(a[:n], b[:n]), true
+	case float64:
+		a, ok := coerce.ToFloat(aVal, super.TypeFloat64)
```

Comment:
This and the other coerce functions here are going to coerce string values to floats which I don't think we want to do here.

Reply:
I know but don't think that matters because the `where` operator containing this filter will drop those values.